src/main/java/org/veupathdb/service/eda/ss/model/variable/VariableWithValues.java

```diff
@@ -36,6 +36,8 @@ public Properties(VariableType type, VariableDataShape dataShape,
     }
   }
 
+  public abstract BinaryProperties getBinaryProperties();
```

Comment:
Consider adding a generic type to this class that is a type that extends BinaryProperties and lives as a field in this superclass.